src/cascadia/TerminalApp/AppKeyBindings.cpp

```diff
@@ -1,158 +1,351 @@
+    DEFINE_EVENT(AppKeyBindings, DecreaseFontSize,  _DecreaseFontSizeHandlers,  TerminalApp::DecreaseFontSizeEventArgs);
+    DEFINE_EVENT(AppKeyBindings, ScrollUp,          _ScrollUpHandlers,          TerminalApp::ScrollUpEventArgs);
+    DEFINE_EVENT(AppKeyBindings, ScrollDown,        _ScrollDownHandlers,        TerminalApp::ScrollDownEventArgs);
+    DEFINE_EVENT(AppKeyBindings, ScrollUpPage,      _ScrollUpPageHandlers,      TerminalApp::ScrollUpPageEventArgs);
+    DEFINE_EVENT(AppKeyBindings, ScrollDownPage,    _ScrollDownPageHandlers,    TerminalApp::ScrollDownPageEventArgs);
+    DEFINE_EVENT(AppKeyBindings, OpenSettings,      _OpenSettingsHandlers,      TerminalApp::OpenSettingsEventArgs);
+
+    // Method Description:
+    // - Deserialize an AppKeyBindings from the key mappings that are in the
+    //   array `json`. The json array should contain an array of objects with
+    //   both a `command` string and a `keys` array, where `command` is one of
+    //   the names listed in `commandNames`, and `keys` is an array of
+    //   keypresses. Currently, the array should contain a single string, which
+    //   can be deserialized into a KeyChord.
+    // Arguments:
+    // - json: and array of JsonObject's to deserialize into our _keyShortcuts mapping.
+    // Return Value:
+    // - the newly constructed AppKeyBindings object.
+    TerminalApp::AppKeyBindings AppKeyBindings::FromJson(const JsonArray& json)
+    {
+        TerminalApp::AppKeyBindings newBindings{};
+
+        for (const auto& value : json)
+        {
+            if (value.ValueType() == JsonValueType::Object)
+            {
+                JsonObject obj = value.GetObjectW();
+                if (obj.HasKey(COMMAND_KEY) && obj.HasKey(KEYS_KEY))
+                {
+                    const auto commandString = obj.GetNamedString(COMMAND_KEY);
+                    const auto keys = obj.GetNamedArray(KEYS_KEY);
+                    if (keys.Size() != 1)
+                    {
+                        continue;
+                    }
+                    const auto keyChordString = keys.GetAt(0).GetString();
+                    ShortcutAction action;
+
+                    // Try matching the command to one we have
+                    if (commandNames.find(commandString) != commandNames.end())
+                    {
+                        action = commandNames.at(commandString);
+                    }
+                    else
+                    {
+                        continue;
+                    }
+
+                    // Try parsing the chord
+                    try
+                    {
+                        auto chord = KeyChordSerialization::FromString(keyChordString);
+                        newBindings.SetKeyBinding(action, chord);
+                    }
+                    catch (...)
+                    {
+                        continue;

```

Comment:
perhaps pop the todo about error handling in here so a code sweep finds it

Reply:
Moved this to the error dialog PR